components/raftstore/src/store/fsm/store.rs

```diff
@@ -1525,19 +1518,44 @@ impl<'a, EK: KvEngine, ER: KvEngine, T: Transport, C: PdClient> StoreFsmDelegate
             return Ok(());
         }
         let check_msg_status = self.check_msg(&msg)?;
-        match check_msg_status {
+        let is_first_request_vote = match check_msg_status {
             CheckMsgStatus::DropMsg => return Ok(()),
-            CheckMsgStatus::PeerExist => (),
+            CheckMsgStatus::FirstRequestVote => true,
             CheckMsgStatus::NewPeer | CheckMsgStatus::NewPeerFirst => {
                 if !self.maybe_create_peer(
                     region_id,
                     &msg,
                     check_msg_status == CheckMsgStatus::NewPeerFirst,
                 )? {
-                    return Ok(());
+                    if !util::is_first_vote_msg(msg.get_message()) {
+                        // Can not create peer from the message and it's not the
+                        // first request vote message.
```

Comment:
Maybe we should add `self.ctx.raft_metrics.message_dropped.region_nonexistent += 1;` to this line and L1546?

Reply:
In fact, this message is not dropped. 🤔